core/src/main/java/com/linecorp/armeria/internal/common/circuitbreaker/CircuitBreakerMappingUtil.java

```diff
@@ -47,5 +51,14 @@ public static String path(ClientRequestContext ctx) {
         return request == null ? "" : request.path();
     }
 
+    @Nullable
+    public static String connectionId(ClientRequestContext ctx) {
+        if (ctx.log().isAvailable(RequestLogProperty.SESSION)) {
+            return Objects.requireNonNull(ctx.log().partial().channel()).id().asLongText();
+        }
+
+        return null;
+    }
+
```

Comment:
Hi @ikhoon , I'm trying to figure out the way to retrieve connection information(netty's `Channel` or armeria's `PooledChannel`) when `CircuitBreakerClientHandler#tryRequest` is called. 

Above method(`connectionId`) doesn't work because the session is not ready(thereby not able to retrieve the channel) when `CircuitBreakerClientHandler#tryRequest` is called as you can see [here](https://github.com/line/armeria/blob/fea0622618613631735b4625c3bf8faf20d9f87a/core/src/main/java/com/linecorp/armeria/client/circuitbreaker/AbstractCircuitBreakerClient.java#L133-L147). 

The order in which `AbstractCircuitBreakerClient` works is as follows: 
1. `final CircuitBreakerCallback callback = handler.tryRequest(ctx, req);`. Can't know the connection information. 
2. `execute` or `doExecute` method is called. The connection is created here.

Reply:
There seems to be `CircuitBreakerCallback` which I might be utilize in this case. 😄